clients/ui/frontend/src/app/pages/modelCatalog/utils/hardwareConfigurationFilterUtils.ts

```diff
@@ -0,0 +1,153 @@
+};
+
+/**
+ * Maps metric and percentile combination to the corresponding artifact field
+ */
+const getLatencyFieldName = (
+  metric: string,
+  percentile: string,
+): keyof PerformanceMetricsCustomProperties => {
+  const metricPrefix = metric.toLowerCase();
+  const percentileSuffix = percentile === 'Mean' ? '_mean' : `_${percentile.toLowerCase()}`;
+  const fieldName = `${metricPrefix}${percentileSuffix}`;
+
+  // Validate that the field exists in PerformanceMetricsCustomProperties
+  // Get latency fields from LatencyMetricKey enum
+  const validFields = Object.values(LatencyMetricKey);
+
+  return validFields.some((field) => field === fieldName)
+    ? // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+      (fieldName as keyof PerformanceMetricsCustomProperties)
+    : LatencyMetricKey.TTFT_MEAN; // Default fallback
+};
+
+/**
+ * Extracts unique hardware types from performance artifacts
+ */
+export const getUniqueHardwareTypes = (
+  artifacts: CatalogPerformanceMetricsArtifact[],
+): string[] => {
+  const hardwareTypes = artifacts
+    .map((artifact) =>
+      getStringValue(artifact.customProperties, ModelCatalogStringFilterKey.HARDWARE_TYPE),
+    )
+    .filter((hardware): hardware is string => !!hardware);
+
+  return [...new Set(hardwareTypes)].toSorted();
+};
+
+/**
+ * Enhanced filter for Max Latency that supports metric and percentile selection
+ */
+export const applyMaxLatencyFilter = (
+  artifact: CatalogPerformanceMetricsArtifact,
+  config: LatencyFilterConfig,
+): boolean => {
+  const fieldName = getLatencyFieldName(config.metric, config.percentile);
+  const latencyValue = getDoubleValue(artifact.customProperties, fieldName);
+  return latencyValue <= config.value;
+};
+
+/**
+ * Filters hardware configuration artifacts based on current filter state
+ */
+export const filterHardwareConfigurationArtifacts = (
+  artifacts: CatalogPerformanceMetricsArtifact[],
+  filterState: ModelCatalogFilterStates,
+): CatalogPerformanceMetricsArtifact[] =>
+  artifacts.filter((artifact) => {
+    // Hardware Type Filter (using central filter state)
+    const hardwareTypeFilters = filterState[ModelCatalogStringFilterKey.HARDWARE_TYPE];
+    if (hardwareTypeFilters.length > 0) {
+      const hardwareType = getStringValue(
+        artifact.customProperties,
+        ModelCatalogStringFilterKey.HARDWARE_TYPE,
+      );
+      if (!hardwareType || !hardwareTypeFilters.includes(hardwareType)) {
+        return false;
+      }
+    }
+
+    // Min RPS Filter
+    const minRpsFilter = filterState[ModelCatalogNumberFilterKey.MIN_RPS];
+    if (minRpsFilter !== undefined) {
+      const totalRps = getTotalRps(artifact.customProperties);
+      if (totalRps < minRpsFilter) {
+        return false;
+      }
+    }
+
+    // Max Latency Filter (using MAX_LATENCY key for now)
+    // TODO: This currently uses TTFT Mean as default - should be enhanced to use
+    // the specific metric/percentile selected by the user
```

Comment:
@manaswinidas I also missed this - editing my review comment above, seems like we need to fix this?